tests/include/test/helpers.h

```diff
@@ -70,19 +70,42 @@ extern mbedtls_test_info_t mbedtls_test_info;
 int mbedtls_test_platform_setup( void );
 void mbedtls_test_platform_teardown( void );
 
+/**
+ * \brief           Record the given, usually current, test as a failure.
+ *
+ * \param test      Name of the test to fail.
```

Comment:
There's some confusion here, partly due to the word “test” being overloaded. There are _test suites_ (concretely, a `.data` file) containing _test cases_ (an entry in a `.data` file). Each test case executes a _test function_ (from the corresponding `.function` file) with certain parameters. Inside the function, there are a number of _test assertions_, usually of the form `TEST_ASSERT` or some other macro.

`test_fail` marks the current _test case_ as failed. There's no usually about it.

The `test` parameter is a string which is displayed to explain the failure. It's typically a stringification of the _test assertion_ that caused the test to be marked as failed. This is not the name of the test case.

The same goes for `skip`, except that there it's a test assumption rather than a test assertion.

Some more important pieces of information about this function:

* If the test case was already marked as failed, the recorded failure information is not overwritten. (I don't recall offhand if this applies to `test_skip` as well: calling `test_skip` twice isn't something we do)
* `test` must be a string literal, because this function stores the pointer that it's given and the pointer must remain valid after the test function returns.
* This function is usually called via macros such as `TEST_ASSERT`, `TEST_EQUAL`, `PSA_ASSERT`, etc.

Reply:
Done. Thanks for the tips on writing documentation, it really helped make it clear where I was going wrong.